Updated configurations for Wire-Cell noise filtering

ebelchio12 opened the pull request:
## Description 
Updated configurations for Wire-Cell noise filtering. The updates in this PR are the new overall response used for signal protection in coherent noise filtering (consistent with SBND field response and electronics response we are currently using. Previous was consistent with ProtoDUNE one), and separate NF configurations for data/MC. There is also a new bad channel masking (updated list of bad channels) and RC response deconvolution in MC (included in noise filtering step). The changes are all about signal processing stage but for data it is separated in reco1 stage, while for MC it is unified with detsim stage.

1. To have WireCell reflect the changes in the wire gdml, we need to update sbnd-wires-geometry-v0*.json.bz2 in the **sbnd_data** directory. The new geometry file can be found here: /exp/sbnd/data/users/lynnt/wirecell/geometry/new_geom/sbnd-wires-geometry-v0202.json.bz2

2. It should go inside the WireCell directory in the corresponding **sbnd_data** version.

bear-is-asleep commented:
> ## Description
> Updated configurations for Wire-Cell noise filtering

Can you provide a more detailed description about what is changed? Also please add a label, i.e. if this affects reco1/reco2, simulation, etc. Thanks!

weihythu reviewed (changes requested):
chndb-response.jsonnet updates look not right. The amplitude of the overall response is too small. Please check your previous plots of updated overall response.

ebelchio12 commented:
@weihythu chndb-resp.jsonnet is now updated.

bear-is-asleep commented:
> @bear-is-asleep the updates are the new response used for signal protection in coherent noise filtering (consistent with SBND field response and electronics response we are currently using. Previous was consistent with ProtoDUNE one), and separate NF configurations for data/MC. There is also a new bad channel masking (updated list of bad channels) and RC response deconvolution in MC (included in noise filtering step). The changes are all about signal processing stage but for data it is separated in reco1 stage, while for MC it is unified with detsim stage.

Thanks! I'm sorry to be pesky but can you edit your description directly? It'll be useful for future collaborators wanting to quickly look at the description. Thanks again!

weihythu reviewed (approved):
Looks good to me.

bear-is-asleep commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/chndb-base.jsonnet:
```diff
@@ -21,14 +21,23 @@ function(params, anode, field, n, rms_cuts=[])
+    // So in total, there are 70 bad channels.
+    // 
+    //bad: [],
+    bad: [546, 607] + std.range(3232, 3263) + std.range(4160, 4191) + [4374, 5054, 5225, 5630, 5631, 7163, 8372, 8568, 10006, 10857],
```
You will need to check with PR #499 to see if these are still the channels.

ebelchio12 replied:
thanks! This is updated to be consistent with the gdml.

bear-is-asleep commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/chndb-base.jsonnet:
```diff
@@ -42,7 +51,7 @@ function(params, anode, field, n, rms_cuts=[])
       {
         //channels: std.range(n * 2560, (n + 1) * 2560 - 1),
-        channels: std.range(n * 5632, n * 5632 + 5631)
+        channels: std.range(n * 5632, n * 5632 + 5631),
```
Same here, will need to check with the new gdml.

ebelchio12 replied:
udpated here as well

bear-is-asleep commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/chndb-perfect.jsonnet:
```diff
@@ -21,14 +21,23 @@ function(params, anode, field, n, rms_cuts=[])
     // coherent noise filtering.
     //groups: [std.range(g*48, (g+1)*48-1) for g in std.range(0,171)],
-    groups: [std.range(n * 5632 + g*64, n * 5632 + (g+1)*64-1) for g in std.range(0,87)],
+    groups: [std.range(n * 5632 + g*32, n * 5632 + (g+1)*32-1) for g in std.range(0,175)], // 32 channels added Ewerton 2024-07
```
Check with PR #499 if these are still the correct channels

ebelchio12 replied:
updated.

bear-is-asleep commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/chndb-perfect.jsonnet:
```diff
@@ -21,14 +21,23 @@ function(params, anode, field, n, rms_cuts=[])
+    // So in total, there are 70 bad channels.
+    // 
+    //bad: [],
+    bad: [546, 607] + std.range(3232, 3263) + std.range(4160, 4191) + [4374, 5054, 5225, 5630, 5631, 7163, 8372, 8568, 10006, 10857],
```
Check with PR #499 if these are still the correct channels

ebelchio12 replied:
updated.

bear-is-asleep commented:
Hi @ebelchio12 , I left a few comments, but it looks like the wire numbers may be updated in #499 . Can you ensure that your changes will not be impacted by the gdml changes? If there's no merge issue, I believe we're good to check the CI and merge.

weihythu commented:
@bear-is-asleep Nice question/reminder. Yes, we should update channel indexing to be consistent with the updated GDML as described in PR #499. @ebelchio12 is working on the updates on our side.

bear-is-asleep commented:
> @bear-is-asleep Nice question/reminder. Yes, we should update channel indexing to be consistent with the updated GDML as described in PR #499. @ebelchio12 is working on the updates on our side.

Thanks! You can include it in this PR or #499, just check w/ @henrylay97 or @lkashur if you chose the latter route. I have a slight preference to put it into #499 so all of the updates go in at the same time, but if the updates are included in this PR we'll just have to make sure this goes in after #499.

lynnt20 commented on sbndcode/WireCell/cfg/pgrapher/experiment/sbnd/nf-data.jsonnet:
```diff
@@ -0,0 +1,86 @@
+function(params, anode, chndbobj, n, name='', dft=default_dft)
+  {
+    local single = {
+      type: 'mbOneChannelNoise', // added Ewerton 2024-07-29
```
very minute comment; for the `type` of each of the groupings for the noise filtering, I notice that the names either reference uboone (mb) or protodune (pd). Does it make sense to change the names of these types to be -> sbnd? Or are the `type` names hardcoded to be something specific for the algorithm?

ebelchio12 replied:
Those are names used in Wire-Cell to register all components of a given module under the scope of that module (in this case, a noise filter module). At this date they are hardcoded (not configurable), and for historical reasons they refer either to uboone or protodune because they were created to be used by those experiments.

lynnt20 commented:
comment for documentation: to have WireCell reflect the changes in the wire gdml, we need to update `sbnd-wires-geometry-v0*.json.bz2` in the `sbnd_data` directory.

lynnt20 commented:
Update: the new geometry file can be found here: `/exp/sbnd/data/users/lynnt/wirecell/geometry/new_geom/sbnd-wires-geometry-v0202.json.bz2`

It should go inside the `WireCell` directory in the corresponding `sbnd_data` version.

weihythu reviewed (changes requested):
@ebelchio12 two comments/changes you need:
1. in your bad channel list, 10863 should be 10869; and you need to add the 12 virtual wires in the gap of each collection plane, [4800, 4801, 4802, 4803, 4804, 4805] and [10438, 10439, 10440, 10441, 10442, 10443]; What are 4385 and 5220?  New bad channels? 
2. You need to update line 144 in `params.jsonnet` with the new filename of wire geometry file `sbnd-wires-geometry-v0202.json.bz2`

@lynnt20 I assume you are coordinating with the manager to push updated wirecell geometry file into next sbnd_data release.

ebelchio12 commented:
@weihythu I updated the filename of geometry file earlier this morning. I also updated the missing combs in APA0 to the correct indexes (previously were listed as 4385 and 5220. Now listed as 4374 and 5231, respectively).

weihythu reviewed (approved):
Thanks @ebelchio12. All looks good to me.
@bear-is-asleep This PR is ready to be merged, however, it has to be pushed along with the updated `sbnd_data`. 
We may see errors in CI test if `sbnd_data` is not updated.

bear-is-asleep commented:
> Thanks @ebelchio12. All looks good to me. @bear-is-asleep This PR is ready to be merged, however, it has to be pushed along with the updated `sbnd_data`. We may see errors in CI test if `sbnd_data` is not updated.

So this PR is awaiting these two things before merging?
- [ ] `sbnd_data` update for wirecell info
- [ ] GDML fix from #499 

If so, I'll have to update the product dependencies to depend on a new version of `sbnd_data`, which isn't a big deal but I'll need to update your PR to refelct this.

weihythu commented:
> > Thanks @ebelchio12. All looks good to me. @bear-is-asleep This PR is ready to be merged, however, it has to be pushed along with the updated `sbnd_data`. We may see errors in CI test if `sbnd_data` is not updated.
> 
> So this PR is awaiting these two things before merging?
> 
> * [ ]  `sbnd_data` update for wirecell info
> * [ ]  GDML fix from [Feature/lkashur crt colwires gdml fix #499](https://github.com/SBNSoftware/sbndcode/pull/499)
> 
> If so, I'll have to update the product dependencies to depend on a new version of `sbnd_data`, which isn't a big deal but I'll need to update your PR to refelct this.

Technically, just need `sbnd_data` update. Thanks for your help.

lynnt20 commented:
CI tests passed successfully when run in my local area when including both `feature/ebelchio12_wirecell_nfsp_v2` and `feature/lkashur_crt_colwires_gdml_fix` in local build, as well as the new version of `sbnd_data`. CI output is posted below. 

@bear-is-asleep, New directory for `sbnd_data` is located here, with the appropriate updates to the changelogs: `/exp/sbnd/app/users/lynnt/wirecell/sbnd_data/v01_25_00`

```
Test project /exp/sbnd/app/users/lynnt/wirecell/build_slf7.x86_64
      Start  1: geometry_sbnd
      Start  2: geometry_sbnd_test
      Start  3: geometry_iterator_sbnd_test
      Start  4: geometry_iterator_loop_sbnd_test
      Start  5: dump_channel_map_sbnd_test
      Start  6: SBNDCRTGeometryTest_1
      Start 10: fast_simulation_reconstruction_test_electrons.sh
      Start 11: fast_simulation_reconstruction_test_muons.sh
      Start 12: prodcosmics_cry_test
      Start 13: testFHiCLfiles_sbnd.sh
 1/10 Test  #5: dump_channel_map_sbnd_test .........................   Passed   11.54 sec
 2/10 Test  #3: geometry_iterator_sbnd_test ........................   Passed   11.58 sec
 3/10 Test  #6: SBNDCRTGeometryTest_1 ..............................   Passed   11.58 sec
 4/10 Test  #4: geometry_iterator_loop_sbnd_test ...................   Passed   11.63 sec
 5/10 Test  #2: geometry_sbnd_test .................................   Passed   16.59 sec
 6/10 Test #12: prodcosmics_cry_test ...............................   Passed   21.29 sec
 7/10 Test  #1: geometry_sbnd ......................................   Passed   45.07 sec
 8/10 Test #13: testFHiCLfiles_sbnd.sh .............................   Passed  174.44 sec
 9/10 Test #11: fast_simulation_reconstruction_test_muons.sh .......   Passed  239.86 sec
10/10 Test #10: fast_simulation_reconstruction_test_electrons.sh ...   Passed  245.08 sec

100% tests passed, 0 tests failed out of 10

Label Time Summary:
DEFAULT    = 788.65 sec*proc (10 tests)
RELEASE    = 788.65 sec*proc (10 tests)

Total Test time (real) = 245.15 sec
```

bear-is-asleep commented:
Thanks all! Let me push the `sbnd_data` changes, then we can go from there.

bear-is-asleep commented:
trigger build LArSoft/lar*@LARSOFT_SUITE_v09_91_02

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23594&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase ci_tests SBND on slf7 for e26:prof - ignored failure for unit_test -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/14889&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/14889&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23593&builds=)

bear-is-asleep commented:
trigger build SBNSoftware/sbndcode#499 LArSoft/lar*@LARSOFT_SUITE_v09_91_02

FNALbuild commented:
:x: CI build for SBND Failed at phase ci_tests SBND on slf7 for c14:prof - ignored failure for unit_test -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/14890&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/14890&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23594&builds=)

bear-is-asleep commented:
trigger build SBNSoftware/sbndcode#499 LArSoft/lar*@LARSOFT_SUITE_v09_91_02

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23596&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23595&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for c14:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23598&builds=)

FNALbuild commented:
:heavy_check_mark: CI build for LArSoft Succeeded on slf7 for e26:prof -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app//view_builds/index?offset=0&builds=lar_ci/23597&builds=)

FNALbuild commented:
:x: CI build for SBND Failed at phase ci_tests SBND on slf7 for c14:prof - ignored failure for unit_test -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/14893&builds=)

:rotating_light: For more details about the failed phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/14893&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20c14:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23598&builds=)

FNALbuild commented:
:warning: CI build for SBND Warning at phase ci_tests SBND on slf7 for e26:prof - ignored failure for unit_test -- details available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/view_builds/index?offset=0&builds=sbnd_ci/14892&builds=)

:rotating_light: For more details about the warning phase, check the [ci_tests SBND phase logs](https://dbweb0.fnal.gov/LarCI/app/ns:SBND/build_detail/phase_details?build_id=sbnd_ci/14892&platform=Linux%20slf7&phase=ci_tests&buildtype=slf7%20e26:prof)

parent CI build details are available through the [CI dashboard](https://dbweb0.fnal.gov/LarCI/app/view_builds/index?offset=0&builds=lar_ci/23595&builds=)
